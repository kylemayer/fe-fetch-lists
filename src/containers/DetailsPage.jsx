/* eslint-disable max-len */
import React, { Component } from 'react';
import VillagerDetail from '../components/displays/VillagerDetail';
import { reqVillagerById } from '../services/reqVillagers';

export default class DetailPage extends Component {
  state = {
    loading: true,
    villager: {},
  };

  componentDidMount = async () => {
    const villager = await reqVillagerById(this.props.match.params.villager);
    this.setState({ loading: false, villager });
  };

  render() {
    const { name, image, species, birthday, skill, phrase } = this.state.villager;

    return (
      <div>
        {this.state.loading
          ? (<p>Loading...</p>)
          : (<VillagerDetail
            name={name}
            image={image}
            species={species}
            birthday={birthday}
            skill={skill}
            phrase={phrase}
          />
          )}
      </div>
    );
  }
}
