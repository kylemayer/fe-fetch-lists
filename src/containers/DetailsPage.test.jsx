
import React from 'react';
import { render, screen } from '@testing-library/react';
import DetailsPage from './DetailsPage';
import { MemoryRouter } from 'react-router-dom';


describe('Details Page', () => {

  it('should render the Details page', async () => {
    render(
      <MemoryRouter>
        <DetailsPage />
      </MemoryRouter>
    );

    const ul = await screen.findByRole('list', { name: 'villagers' });
    expect(ul).toBeInTheDocument();
    expect(ul).toMatchSnapshot();
  });
});
