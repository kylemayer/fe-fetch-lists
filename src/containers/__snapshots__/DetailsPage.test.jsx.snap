// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Page should render the List page 1`] = `
<ul
  aria-label="villagers"
>
  <li>
    <a
      href="/1"
    >
      <h3>
        Admiral
      </h3>
      <img
        alt="Admiral"
        class="Villager"
        src="example.com/image.png"
      />
      <p>
        bird
      </p>
      <p>
        January 27th (Aquarius)
      </p>
      <p>
        Writing about pickles
      </p>
      <p>
        aye aye
      </p>
    </a>
  </li>
</ul>
`;
